File: leetcode/enumerated/1.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        diffs = {}
        for i in range(len(nums)):            
            if nums[i] in diffs:
                return [diffs[nums[i]], i]
            
            diffs[target - nums[i]] = i

File: leetcode/enumerated/cpp/1.cpp
// https://leetcode.com/problems/two-sum/
#include <map>
#include <vector>

using std::vector;
using std::map;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> diffs;
        for (int i = 0; i < nums.size(); i++) {
            if (diffs.find(nums[i]) == diffs.end()) {
                diffs[target - nums[i]] = i;
            } else {
                return vector<int>{diffs[nums[i]], i};
            }
        }
    }
};
